test(snapshot/clone): add bdd for create and list volume from snapshot

tiagolobocastro opened the pull request:
Adds BDD specification and respective tests.

tiagolobocastro committed:
test(snapshot/clone): add bdd for create and list volume from snapshot

Adds BDD specification and respective tests.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro commented:
@hrudaya21 any other scenarios missing?

hrudaya21 commented on tests/bdd/features/snapshot/clone/create_parameters.feature:
```diff
@@ -0,0 +1,24 @@
+  # This may be allowed in the future by inflating the clones
+  Scenario: Thick provisioning
+    When the request is for a thick volume
+    Then the request should fail with InvalidArguments
```
Whether these checks are part of Control Plane ? As part of clone create gRPC, We don't have the parameters for clone size and thick/thin provision parameters.

tiagolobocastro replied:
Yes these are control-plane checks, because create volume from snapshot has the same parameters from csi, so we just validate they are correct.

tiagolobocastro commented:
bors try

tiagolobocastro commented:
bors cancel

Abhinandan-Purkait commented on tests/bdd/features/snapshot/clone/create.feature:
```diff
@@ -0,0 +1,14 @@
+
+  Scenario: Create multiple new volumes as snapshot clones for a valid snapshot
+    When we attempt to create 4 new volumes with the snapshot as their source
+    Then all requests should succeed
```
This one is good.

Can we also add a test case where we attempt continuous snapping and cloning from the newly created clones.

For ex, 
Vol 1 -> Snap 1 -> Clone 1
Clone 1 -> Snap 2 -> Clone 2
Clone 2 -> Snap 3 -> Clone 3

tiagolobocastro replied:
This is nice! Yes I'll do it

tiagolobocastro replied:
Done :)

tiagolobocastro commented:
bors merge

tiagolobocastro committed:
refactor(snapshot/clone): less liberal use of clone

Cloning on CSI terms refers to clone a volume from another volume.
Particularly on k8s, creating a volume from a snapshot is referred to as
restoring a snapshot into a new volume.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro commented:
bors merge

tiagolobocastro committed:
ci(bors): undo staging tests

This may explain some issues we see when PR's don't get merged?

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro commented:
bors cancel

tiagolobocastro commented:
bors merge

tiagolobocastro commented:
bors r-

tiagolobocastro commented:
bors cancel

tiagolobocastro commented:
bors ping

tiagolobocastro commented:
bors merge

tiagolobocastro commented:
bors merge
